test/snippet/index/bi_fm_index_iterator.cpp

```diff
@@ -0,0 +1,92 @@
+#include <vector>
+#include <iostream>
+#include <seqan3/index/all.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+int main()
+{
+    {
+        //! [cycle]
+        std::vector<dna4> genome{"GAATTAATGAAC"_dna4};
+        bi_fm_index<std::vector<dna4>> index{genome}; // build the index
+
+        // TODO(comments): outputs [A, A, G] instead of AAG
+
+        auto it = index.begin(); // create an iterator
+        // it.cycle_back(); // cycle_back / cycle_front on begin() is undefined behaviour!
+        it.extend_right("AAC"_dna4); // search the sequence "AAC"
+        std::cout << it.query() << '\n'; // outputs "AAC"
+        std::cout << it.last_char() << '\n'; // outputs 'C'
+
+        // it.cycle_front(); // undefined behaviour! only cycle_back() is allowed after extend_right()
```

Comment:
Would it be better to throw? Does one cycle often in algorithms?

Reply:
I think we already discussed this and decided to make it undefined behaviour. Yes, pretty much all the time when you do approximate string matching